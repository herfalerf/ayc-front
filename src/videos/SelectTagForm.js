import React from "react";
import { Field, Form, Formik } from "formik";
import { Button } from "@material-ui/core";
import * as yup from "yup";

const SelectTagForm = ({ setVideo, videoId, availableTags, tagVideo }) => {
  console.debug("TagVideoForm", "videoId", videoId);
  const schema = yup.object().shape({
    tag: yup.string().required("Tag is required"),
  });
  return (
    <div>
      <h2>Add a Tag</h2>
      <Formik
        validationSchema={schema}
        initialValues={{ tag: undefined }}
        onSubmit={async (values, { setStatus, setSubmitting }) => {
          setStatus(undefined);
          console.debug("videoId=", videoId, "values=", values);
          let data = { tag_id: values.tag };
          let result = await tagVideo(videoId, data);
          console.debug(result);

          if (result.success) {
            setSubmitting(false);
            setVideo(result.video);
          } else {
            setStatus({ error: result.errors });
          }
        }}
      >
        {({ status }) => (
          <Form>
            <Field as="select" name="tag">
              <option value={undefined}>Select a tag</option>

              {availableTags &&
                availableTags.map((t) => (
                  <option value={t.id} label={t.name} key={t.name} />
                ))}
            </Field>
            <br />
            {status && status.error ? (
              <div>API Error: {status.error}</div>
            ) : null}
            <br />
            <Button variant="contained" color="primary" type="submit">
              Add Tag
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default SelectTagForm;
