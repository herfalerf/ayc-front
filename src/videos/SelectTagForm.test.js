import React from "react";
import { render } from "@testing-library/react";
import SelectTagForm from "./SelectTagForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <SelectTagForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <SelectTagForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
