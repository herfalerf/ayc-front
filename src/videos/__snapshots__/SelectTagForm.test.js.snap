// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Add a Tag
    </h2>
    <form
      action="#"
    >
      <select
        name="tag"
      >
        <option>
          Select a tag
        </option>
      </select>
      <br />
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Add Tag
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
